src/xoreostex2tga.cpp

```diff
@@ -194,4 +196,14 @@ void convert(const Common::UString &inFile, const Common::UString &outFile,
 		image->flipVertically();
 
 	image->dumpTGA(outFile);
+
+	std::unique_ptr<Common::SeekableReadStream> txiData(image->getTXI());
+	if (txiData != nullptr) {
+		Common::WriteFile f(Common::FilePath::changeExtension(outFile, "txi"));
+		for (size_t iByte = 0; iByte < txiData->size(); iByte++) {
+			f.writeByte(txiData->readByte());
+		}
```

Comment:
Copying data byte by byte probably not the best way, I suspect that performance would be worst. Maybe there has a bulk API?

Reply:
Maybe indeed. Although I have no idea which method to call in order to write the whole chunk of data at once

Reply:
You can throw the `txiData` `SeekableWriteStream` directly into `Common::WriteStream::writeStream(ReadStream &stream)` ([src/common/writestream.h:211)](https://github.com/xoreos/xoreos-tools/blob/master/src/common/writestream.h#L211)

Reply:
(I.e. `Common::WriteFile` is a `Common::WriteStream`, so `f.writeStream(*txiData)` should work)